src/backend/MediatypeS11n.cc

```diff
@@ -0,0 +1,89 @@
+//
+//  backend/MediatypeS11n.cc
+//  apib::backend
+//
+//  Created by Thomas Jandecka on 02/21/2020
+//  Copyright (c) 2020 Apiary Inc. All rights reserved.
+//
+
+#include "MediatypeS11n.h"
+
+#include <cctype>
+// FIXME: avoid relative, cross-modular include by setting up build environment
+//          to search for includes in `src`; then `#include <parser/Mediatype.h>`
+#include "../parser/Mediatype.h"
+
+using namespace apib::backend;
+
+namespace
+{
+    std::ostream& serialize_quote_escaped(std::ostream& out, const std::string& v)
+    {
+        for (const auto& c : v)
+            if (c == '"')
+                out << "\\\"";
+            else
+                out << c;
+        return out;
+    }
+
+    std::ostream& serialize_quoted(std::ostream& out, const std::string& v)
+    {
+        return serialize_quote_escaped(out << '"', v) << '"';
+    }
+
+    bool is_restricted(char c) noexcept
+    {
+        for (const auto& r : { '!', '#', '$', '&', '^', '_', '-', '.' })
```

Comment:
Are you working from RFC 6838 or a different one?

I think we might be missing `+` perhaps `restricted-name-chars =/ "+"`? https://tools.ietf.org/html/rfc6838#section-4.2

Reply:
really good catch :clap: